components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -85,16 +92,20 @@ object typing {
   case class TypedTaggedValue(underlying: SingleTypingResult, tag: String) extends TypedObjectWithData {
     override def data: String = tag
 
+    override def value: Option[Any] = underlying.value
+
     override def withoutValue: TypedTaggedValue = TypedTaggedValue(underlying.withoutValue, tag)
 
     override def display: String = s"${underlying.display} @ $tag"
   }
 
-  case class TypedObjectWithValue private[typing](underlying: TypedClass, value: Any) extends TypedObjectWithData {
+  case class TypedObjectWithValue private[typing](underlying: TypedClass, justValue: Any) extends TypedObjectWithData {
```

Comment:
I think that we rather use "opt"/"optional". Also value should be rather `valueOpt` instead of changing of this field. Be aware that this is part of our core API and each change will need migration guide entry